crates/relayer-framework-test/src/relayer_mock/contexts/chain.rs

```diff
@@ -0,0 +1,236 @@
+//! The MockChainContext uses a ChainState alias for it's consensus_state.
+//! This type is a HashMap<Height, State>, used to represent the State of
+//! the chain at all Heights.
+//! The consensus_states is a HashMap<Height, ChainState> used to send the
+//! consensus_state of the chain at a specific Height inside a client upgrade
+//! message.
+//! The client_states is a HashMap<ClientId, HashMap<Height, ChainState>>.
+//! This is used to check the consensus_state for a specific client, at a
+//! specific height.
+//! Usually the client_states would use the root hashes of a Merle Tree,
+//! but since the MockChain is used for testing and will have a small number
+//! of states, the whole state is used. This avoids needing to implement
+//! Merkle Trees and Proofs.
+
+use alloc::string::String;
+use std::collections::hash_map::Entry;
+use std::sync::Mutex;
+use std::{collections::HashMap, sync::Arc};
+
+use crate::relayer_mock::base::error::Error;
+use crate::relayer_mock::base::types::aliases::{
+    ChainState, ChannelId, ClientId, ConsensusState, PortId, Sequence,
+};
+use crate::relayer_mock::base::types::runtime::{MockChainRuntimeContext, MockRuntimeContext};
+use crate::relayer_mock::base::types::{
+    height::Height as MockHeight, packet::PacketKey, state::State,
+};
+use ibc_relayer_framework::base::one_for_all::traits::runtime::OfaRuntimeContext;
+
+pub struct MockChainContext {
+    pub name: String,
+    pub consensus_states: Arc<Mutex<HashMap<MockHeight, ChainState>>>,
+    pub consensus_state: Arc<Mutex<ChainState>>,
```

Comment:
The consensus state refers to the state of other chains recorded on the current chain. For the current chain's state, we can just call it `current_state`. It looks like the `consensus_states` field here should actually be called `past_chain_states`.